runtime/runtime-standalone/src/lib.rs

```diff
@@ -0,0 +1,394 @@
+    pub genesis_time: u64,
+    pub gas_price: Balance,
+    pub gas_limit: Gas,
+    pub runtime_config: RuntimeConfig,
+    pub state_records: Vec<StateRecord>,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct Block {
+    prev_block: Option<Box<Block>>,
+    state_root: CryptoHash,
+    gas_burnt: Gas,
+    pub epoch_height: EpochHeight,
+    pub block_height: BlockHeight,
+    pub block_timestamp: u64,
+    pub gas_price: Balance,
+    pub gas_limit: Gas,
+}
+
+impl Block {
+    pub fn genesis(genesis_config: &GenesisConfig) -> Self {
+        Self {
+            prev_block: None,
+            state_root: CryptoHash::default(),
+            block_height: 1,
+            epoch_height: 1,
+            block_timestamp: genesis_config.genesis_time,
+            gas_price: genesis_config.gas_price,
+            gas_limit: genesis_config.gas_limit,
+            gas_burnt: 0,
+        }
+    }
+
+    pub fn produce(&self, new_state_root: CryptoHash) -> Block {
+        Self {
+            gas_price: self.gas_price,
+            gas_limit: self.gas_limit,
+            block_timestamp: self.block_timestamp + 1,
+            prev_block: Some(Box::new(self.clone())),
+            state_root: new_state_root,
+            block_height: self.block_height + 1,
+            epoch_height: self.epoch_height + 1, // TODO: simulate epoch_height
+            gas_burnt: 0,
+        }
+    }
+}
+
+pub struct RuntimeStandalone {
+    tx_pool: TransactionPool,
+    transactions: HashMap<CryptoHash, SignedTransaction>,
+    outcomes: HashMap<CryptoHash, ExecutionOutcome>,
+    cur_block: Block,
+    runtime: Runtime,
+    trie: Arc<Trie>,
+    pending_receipts: Vec<Receipt>,
+}
+
+impl RuntimeStandalone {
+    pub fn new(genesis: GenesisConfig, store: Arc<Store>) -> Self {
+        let mut genesis_block = Block::genesis(&genesis);
+        let mut store_update = store.store_update();
+        let runtime = Runtime::new(genesis.runtime_config.clone());
+        let trie = Arc::new(Trie::new(store));
+        let trie_update = TrieUpdate::new(trie.clone(), CryptoHash::default());
+        let (s_update, state_root) =
+            runtime.apply_genesis_state(trie_update, &[], &genesis.state_records);
+        store_update.merge(s_update);
+        store_update.commit().unwrap();
+        genesis_block.state_root = state_root;
+        Self {
+            trie,
+            runtime,
+            transactions: HashMap::new(),
+            outcomes: HashMap::new(),
+            cur_block: genesis_block,
+            tx_pool: TransactionPool::new(),
+            pending_receipts: vec![],
+        }
+    }
+
+    /// Processes blocks until the final value is produced
+    pub fn resolve_tx(
+        &mut self,
+        mut tx: SignedTransaction,
+    ) -> Result<ExecutionOutcome, RuntimeError> {
+        tx.init();
+        let mut outcome_hash = tx.get_hash();
+        self.transactions.insert(outcome_hash, tx.clone());
+        self.tx_pool.insert_transaction(tx);
```

Comment:
it seems that `send_tx` can be used here.

Reply:
Right, I was considered it, but decided to be explicit there.

Reply:
Sending a tx before execution is explicit enough, in my opinion.